src/main/java/com/iota/iri/utils/dag/TraversalException.java

```diff
@@ -0,0 +1,38 @@
+package com.iota.iri.utils.dag;
+
+/**
+ * This class is used to wrap exceptions that are specific to traversing the graph.
+ *
+ * It allows us to distinct between the different kinds of errors that can happen during the execution of the code.
+ */
+public class TraversalException extends Exception {
+    /**
+     * Constructor of the exception which allows us to provide a specific error message and the cause of the error.
+     *
+     * @param message reason why this error occured
+     * @param cause wrapped exception that caused this error
+     */
```

Comment:
Wow such detailed javadocs :-)

I would have been lazy and just used the `@see` but I will not complain :-)